File: MaleFashion.Server/Models/Entities/ProductVariant.cs
using System.Drawing;

namespace MaleFashion.Server.Models.Entities
{
    public class ProductVariant
    {
        public int Id { get; set; }
        public int Stock { get; set; }

        public int ProductId { get; set; }
        public virtual Product? Product { get; set; }

        public int ColorId { get; set; }
        public virtual Color? Color { get; set; }

        public int SizeId { get; set; }
        public virtual Size? Size { get; set; }

        public virtual ICollection<CartItem>? CartItems { get; set; }
        public virtual ICollection<OrderItem>? OrderItems { get; set; }
        public virtual ICollection<Favorite>? Favorites { get; set; }
    }
}

File: malefashion.client/src/models/entities/ProductVariant.ts
import { Color } from "./Color";
import { Product } from "./Product";
import { Size } from "./Size";

export interface ProductVariant {
    id: number;
    stock: number;

    productId: number;
    product?: Product;

    colorId?: number;
    color?: Color;

    sizeId?: number;
    size?: Size;
}